ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -686,11 +686,14 @@ class JITModBuilder {
 
     // Make more space to store the locations of the corresponding live AOT
     // values for this guard failure.
-    size_t Offset = LiveAOTSize;
-    LiveAOTSize += sizeof(AOTInfo) * LiveValues.size();
-    LiveAOTAddr = (AOTInfo *)realloc(LiveAOTAddr, LiveAOTSize);
+    size_t CurPos = LiveAOTNum;
+    LiveAOTNum += LiveValues.size();
+    LiveAOTArray = (AOTInfo *)reallocarray(LiveAOTArray, LiveAOTNum, sizeof(AOTInfo));
+    if (LiveAOTArray == NULL) {
```

Comment:
This can just be an `assert` IMHO since we don't appear to be able to recover from an error.

Reply:
Fixed in [a52ef2a](https://github.com/ykjit/yk/pull/565/commits/a52ef2afdae235c5ef07bfb6ba44e2631607ec16)